Failing with runtime error: invalid memory address or nil pointer dereference on Packer 1.1.1

Packer Version - 1.1.1
Host Platform - 4.4.51-40.58.amzn1.x86_64 #1 SMP Tue Feb 28 21:57:17 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- Debug log output
https://gist.github.com/shankarb01/b04044cc6b67c54005f780bb869e5668

We are using the same template successfully with a very old version of Packer i.e 0.10.1 but decided to upgrade to the latest binary of Packer as we needed to use config parameters `temporary_security_group_source_cidr` and `snapshot_tags` which i believe are introduced as part of 1.1.1.

I've seen a couple of these as well, but still looking for a way to reproduce

```
panic: runtime error: invalid memory address or nil pointer dereference
2017/10/27 14:25:05 packer: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x16de4dd]
2017/10/27 14:25:05 packer:
2017/10/27 14:25:05 packer: goroutine 23 [running]:
2017/10/27 14:25:05 packer: github.com/hashicorp/packer/vendor/github.com/aws/aws-sdk-go/aws/request.Waiter.WaitWithContext(0x2c6ab0a, 0x16, 0xc4206a02d0, 0x3, 0x3, 0x3cce380, 0xc42016c010, 0x28, 0x0, 0x0, ...)
2017/10/27 14:25:05 packer: 	/private/tmp/packer-20171014-45705-5rw8o9/src/github.com/hashicorp/packer/vendor/github.com/aws/aws-sdk-go/aws/request/waiter.go:195 +0x18d
2017/10/27 14:25:05 packer: github.com/hashicorp/packer/builder/amazon/common.waitUntilSecurityGroupExists(0xc42016c088, 0xc4202921e0, 0xc4201076c8, 0x1)
2017/10/27 14:25:05 packer: 	/private/tmp/packer-20171014-45705-5rw8o9/src/github.com/hashicorp/packer/builder/amazon/common/step_security_group.go:194 +0x262
2017/10/27 14:25:05 packer: github.com/hashicorp/packer/builder/amazon/common.(*StepSecurityGroup).Run(0xc42054c050, 0x3ce1a80, 0xc4205d6990, 0x0)
2017/10/27 14:25:05 packer: 	/private/tmp/packer-20171014-45705-5rw8o9/src/github.com/hashicorp/packer/builder/amazon/common/step_security_group.go:109 +0xa37
2017/10/27 14:25:05 packer: github.com/hashicorp/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc4203f5180, 0x3ce1a80, 0xc4205d6990)
2017/10/27 14:25:05 packer: 	/private/tmp/packer-20171014-45705-5rw8o9/src/github.com/hashicorp/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x1ee
2017/10/27 14:25:05 packer: github.com/hashicorp/packer/builder/amazon/ebs.(*Builder).Run(0xc4200baf00, 0x3ce4aa0, 0xc4200e0320, 0x3cdbcc0, 0xc420164400, 0x3ce3400, 0xc42016c008, 0x80000000018, 0x28, 0x28, ...)
2017/10/27 14:25:05 packer: 	/private/tmp/packer-20171014-45705-5rw8o9/src/github.com/hashicorp/packer/builder/amazon/ebs/builder.go:248 +0x153c
2017/10/27 14:25:05 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc4200e10a0, 0x1, 0xc420016150, 0x0, 0x0)
2017/10/27 14:25:05 packer: 	/private/tmp/packer-20171014-45705-5rw8o9/src/github.com/hashicorp/packer/packer/rpc/builder.go:93 +0x1dd
2017/10/27 14:25:05 packer: reflect.Value.call(0xc42006a840, 0xc42000e540, 0x13, 0x2c463d2, 0x4, 0xc42040df20, 0x3, 0x3, 0xc42006bf78, 0xc420023c78, ...)
2017/10/27 14:25:05 packer: 	/usr/local/Cellar/go/1.9.1/libexec/src/reflect/value.go:434 +0x906
2017/10/27 14:25:05 packer: reflect.Value.Call(0xc42006a840, 0xc42000e540, 0x13, 0xc42048a720, 0x3, 0x3, 0xc42048a760, 0x131b0f5, 0xc420023c20)
2017/10/27 14:25:05 packer: 	/usr/local/Cellar/go/1.9.1/libexec/src/reflect/value.go:302 +0xa4
2017/10/27 14:25:05 packer: net/rpc.(*service).call(0xc420065d00, 0xc420088640, 0xc420017130, 0xc4200fa700, 0xc4200e13a0, 0x26c5420, 0xc42001612c, 0x18a, 0x266c6e0, 0xc420016150, ...)
2017/10/27 14:25:05 packer: 	/usr/local/Cellar/go/1.9.1/libexec/src/net/rpc/server.go:381 +0x142
2017/10/27 14:25:05 packer: created by net/rpc.(*Server).ServeCodec
2017/10/27 14:25:05 packer: 	/usr/local/Cellar/go/1.9.1/libexec/src/net/rpc/server.go:475 +0x36b
2017/10/27 14:25:05 [INFO] (telemetry) ending amazon-ebs
2017/10/27 14:25:05 [INFO] (telemetry) found error: unexpected EOF
2017/10/27 14:25:05 ui error: Build 'amazon-ebs' errored: unexpected EOF
2017/10/27 14:25:05 Builds completed. Waiting on interrupt barrier...
2017/10/27 14:25:05 machine readable: error-count []string{"1"}
2017/10/27 14:25:05 ui error:
==> Some builds didn't complete successfully and had errors:
2017/10/27 14:25:05 machine readable: amazon-ebs,error []string{"unexpected EOF"}
2017/10/27 14:25:05 ui error: --> amazon-ebs: unexpected EOF
2017/10/27 14:25:05 ui:
==> Builds finished but no artifacts were created.
2017/10/27 14:25:05 [INFO] (telemetry) Finalizing.
2017/10/27 14:25:05 /usr/local/bin/packer: plugin process exited
2017/10/27 14:25:05 waiting for all plugin processes to complete...
2017/10/27 14:25:05 /usr/local/bin/packer: plugin process exited
2017/10/27 14:25:05 /usr/local/bin/packer: plugin process exited
2017/10/27 14:25:05 /usr/local/bin/packer: plugin process exited
2017/10/27 14:25:05 /usr/local/bin/packer: plugin process exited
```

--- a/builder/amazon/common/step_security_group.go
+++ b/builder/amazon/common/step_security_group.go
@@ -158,6 +158,7 @@ func waitUntilSecurityGroupExists(c *ec2.EC2, input *ec2.DescribeSecurityGroupsI
 	w := request.Waiter{
 		Name:        "DescribeSecurityGroups",
 		MaxAttempts: 40,
+		Delay:       request.ConstantWaiterDelay(5 * time.Second),
 		Acceptors: []request.WaiterAcceptor{
 			{
 				State:    request.SuccessWaiterState,

